### PR TITLE
rewrote expand_types in Python

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

File: hail/python/hail/table.py [hunk 1]
```diff
@@ -2019,12 +2019,29 @@ def expand_types(self):
         :class:`.Table`
             Expanded table.
         """
+        def _expand(e):
+            if isinstance(e, CollectionExpression) or isinstance(e, DictExpression):
+                return hl.map(lambda x: _expand(x), hl.array(e))
+            elif isinstance(e, StructExpression):
```
Comment (tpoterba) [inline hail/python/hail/table.py]: need tuple too

Comment (cseed) [reply hail/python/hail/table.py]: Good catch, fixed, thanks!

**Comment by cseed** [comment]
I also broke out routines to create examples of values (tables, matrix tables) with all types, and added a test case that runs expand_types on the table of values of all types.

File: hail/python/test/hail/helpers.py [hunk 2]
```diff
@@ -86,3 +86,48 @@ def assert_time(f, max_duration):
...
+    return hl.struct(**{prefix + k: s[k] for k in s})
+
+def create_all_values_table():
+    all_values = create_all_values()
+    return (hl.utils.range_table(5, n_partitions=3)
+            .annotate_globals(**prefix_struct(all_values, 'global_'))
+            .annotate(**all_values)
+            .cache())
+    
+
```
Comment (danking) [inline hail/python/test/hail/helpers.py]: extra space

Comment (danking) [reply hail/python/test/hail/helpers.py]: (aside: we should get flake8 & pylint set up on hail's repo so this stuff can be caught locally in an automated fashion)

**Pull request merged**